### PR TITLE
Simplify BlockPatternsSyncFilter with clearer labels and additional context

### PR DESCRIPTION
**Pull request by richtabor**
## What?
A follow-up to https://github.com/WordPress/gutenberg/pull/54681 that:

- Reduces individual labeling and help text with more context below (similar to the "Tools" control in the block editor header). 
- Renames "Standard" to "Not synced", which adds more context to what these patterns are - and reduces nomenclature. 

Props @jasmussen. 

## Testing Instructions

1. Open a post or page.
2. Open the pattern inserter.
3. Select the "Filter patterns" control.
4. See changes.

### Before

CleanShot 2023-09-26 at 14 21 22

### After

CleanShot 2023-09-26 at 14 20 44

### CODE REVIEW

**Comment by richtabor** [comment]
In the same vein as @aaronrobertshaw's comment [here](https://github.com/WordPress/gutenberg/pull/54807#issuecomment-1734819029), this tweak polishes up the copy to help make patterns more understandable for 6.4. 

Also, as a note — I don't think we should be referencing "core" in interface elements. It can be confusing for folks.

**Review by glendaviesnz — APPROVED** [review]
Tested well for me. @SaxonF do you want to give it a 👀 over from a design perspective?

The only thought I had is that not all the patterns come from the directory, some of them are bundled in the core release I believe, but I don't think that is an important enough distinction to confuse users over - I agree with Rich that the use of the term `core` is not that meaningful to most users.

**Comment by jasmussen** [comment]
I'm a big fan, nice iteration Rich!

**Review by aaronrobertshaw — COMMENTED** [review]
I like the cleaner menu @richtabor, nice work ✨ 

A couple of quick thoughts came to mind.

1. As @glendaviesnz mentioned, the "Pattern directory" option doesn't really cover core patterns, but I also agree with not referencing "core" in the UI. Given this, I wonder if "Pattern directory" narrows expectations too much to only the .org pattern directory? If we are looking for alternatives, in other conversations someone suggested "WordPress" as an option.

2. Something that was already missing from the source filters was that plugins can register patterns as well as themes. Should that be noted in the help text at the bottom of the menu or via the menu items?

Nothing in those thoughts should block this PR but perhaps there is still some further tweaking we can do.

I believe the 6.4 beta 1 has been cut as well, so if this is intended for 6.4 we might need the `Backport to WP Beta/RC` label on this.

**Comment by jasmussen** [comment]
Thanks for the clarifications, just in terms of finding good names here, how are the sources distributed at the moment? It sounds like:

1. Patterns from the pattern directory
2. Patterns bundled with WordPress
3. Patterns bundled with a theme
4. Patterns added by a plugin
5. All

Are those 5 separate filters where 2 of them (from WordPress and from a plugin) are missing? Or are those filters implied in any of the existing filters (such as "Bundled & Pattern Directory", or "Themes & Plugins")?

For what it's worth, removing the help text isn't about making it less verbose, it's about ensuring that the line-height of each item has equal weight, where previously "All" was the only one not having help text and therefore feeling less substantial. Because of that, I think we can definitely be more verbose with the main title of each filter, if we need to.

**Comment by aaronrobertshaw** [comment]
> how are the sources distributed at the moment? It sounds like:

The only one I think is missing from your list @jasmussen is user-created patterns. While they could be considered under "all", I suspect people will want to be able to filter by that source.

> Are those 5 separate filters where 2 of them (from WordPress and from a plugin) are missing? Or are those filters implied in any of the existing filters (such as "Bundled & Pattern Directory", or "Themes & Plugins")?

The current "Directory" filter includes both WordPress (core) and pattern directory patterns. Essentially, patterns that have one of the following sources:

- `core`
- `pattern-directory/core`
- `pattern-directory/featured`
- `pattern-directory/theme`

The "Theme" filter currently includes all patterns that don't come from the above sources and aren't a user-created pattern. So as it stands now, that filter includes both theme and plugin patterns.

> For what it's worth, removing the help text isn't about making it less verbose

Thanks for the clarification 👍

**Comment by mikachan** [comment]
👋 Just a nudge to please merge these changes if you'd like them included in Gutenberg 16.7.

**Comment by jasmussen** [comment]
> The only one I think is missing from your list @jasmussen is user-created patterns. While they could be considered under "all", I suspect people will want to be able to filter by that source.

Ah yes of course.

> The current "Directory" filter includes both WordPress (core) and pattern directory patterns. 
> The "Theme" filter currently includes all patterns that don't come from the above sources and aren't a user-created pattern. So as it stands now, that filter includes both theme and plugin patterns.

Gotcha. I'll defer to Rich on any edits to this PR, but tentatively the list could be this:

```
All ✅
Bundled & Pattern Directory
Themes & Plugins
User
```

**Comment by mikachan** [comment]
As this PR hasn't been merged yet and the latest version of Gutenberg is due to be released today, I'm going to remove the `Backport to Gutenberg RC` label.

I'm going to add the `Backport to WP Beta/RC` label, as from reading the comments above, I believe this work should be included in WordPress 6.4.

**Comment by richtabor** [comment]
> Something that was already missing from the source filters was that plugins can register patterns as well as themes. Should that be noted in the help text at the bottom of the menu or via the menu items?

Perhaps "Theme" could become "Theme & Plugins" (as there's only one active theme)? 

> Gotcha. I'll defer to Rich on any edits to this PR, but tentatively the list could be this:

> All ✅
> Bundled & Pattern Directory
> Themes & Plugins
> User

@jasmussen I'm not following what the difference between "Bundled" and "Themes & Plugins" is? Are"bundled" patterns the curated patterns from the directory?

**Comment by jasmussen** [comment]
> I'm not following what the difference between "Bundled" and "Themes & Plugins" is? Are"bundled" patterns the curated patterns from the directory?

It's my attempt to find a better word than "core". Essentially those that come with the software. It isn't clear to me if we actually need to call that out or not, and I'd be happy to keep the term just "Pattern Directory", mainly responding here to the comments by Aaron.

**Comment by aaronrobertshaw** [comment]
> I'm not following what the difference between "Bundled" and "Themes & Plugins" is? Are"bundled" patterns the curated patterns from the directory?

Within WordPress there are patterns that get registered [directly from files](https://github.com/WordPress/wordpress-develop/tree/a807e86391817a45dd82f54bef234be1a28d4c6d/src/wp-includes/block-patterns) in core. These patterns are **not** in the WordPress.org Pattern Directory. In addition to these "core" patterns, [additional patterns are loaded](https://github.com/WordPress/wordpress-develop/blob/7faab129f56779d0576bed37d025ef16bcb95bf7/src/wp-includes/rest-api/endpoints/class-wp-rest-block-patterns-controller.php#L105-L112) from the pattern directory. 

When you select the "Directory" filter, all of the above patterns are shown. That's why the help text prior to this PR stated "Pattern directory and core".

Themes and plugins can register their own patterns (e.g. [TT3's patterns](https://github.com/WordPress/wordpress-develop/tree/7faab129f56779d0576bed37d025ef16bcb95bf7/src/wp-content/themes/twentytwentythree/patterns)). These sorts of patterns currently show when selecting the "Themes" filter. It's because plugin-registered patterns appear under this filter I noted that its current label is not 100% accurate. 

I only wanted to flag the gap in our terminology here. As @glendaviesnz noted, these finer points might not be worth the distinction in the UI if it could cause further confusion. I'm happy to defer to your collective wisdom here as I wasn't looking to block these improvements at all.

**Comment by jameskoster** [comment]
> If we are looking for alternatives, in other conversations someone suggested "WordPress" as an option.

This might be worth considering given the "Directory" patterns only include those authored by the WordPress.org user (IE the curated category). "Directory" alone could be misleading since community patterns aren't included?

**Comment by richtabor** [comment]
> When you select the "Directory" filter, all of the above patterns are shown. That's why the help text prior to this PR stated "Pattern directory and core".

> Within WordPress there are patterns that get registered [directly from files](https://github.com/WordPress/wordpress-develop/tree/a807e86391817a45dd82f54bef234be1a28d4c6d/src/wp-includes/block-patterns) in core. These patterns are not in the WordPress.org Pattern Directory.

I think it's fine not declaring a difference between core and Pattern Directory. They're loaded from core, but only because we don't have support for blockTypes in the pattern directory yet (otherwise they would be directory). They're _essentially_ directory patterns. 

> This might be worth considering given the "Directory" patterns only include those authored by the WordPress.org user (IE the curated category). "Directory" alone could be misleading since community patterns aren't included?

> If we are looking for alternatives, in other conversations someone suggested "WordPress" as an option.

I'm ok with "WordPress" as a label, though I do think the more specific "Pattern Directory" connects the dots a bit better.

**Comment by jasmussen** [comment]
I don't think "core" as it ships in trunk is meaningful. The Pattern Directory _is_ WordPress.org, and aren't the bundled patterns sourced from there anyway? So to that end, with both the added is simplicity and the help text at the bottom, I think this one is fine to ship as is.

**Comment by richtabor** [comment]
Here's where it's at now. Good to go? 

CleanShot 2023-09-29 at 11 46 44

**Review by jasmussen — APPROVED** [review]

**Comment by SaxonF** [comment]
Nice work @richtabor

**Comment by mikachan** [comment]
I just cherry-picked this PR to the 6.4-beta3 branch to get it included in the next release: 05755f32a9

**Pull request merged**
